efi/preinstall/checks_context.go

```diff
@@ -0,0 +1,585 @@
+		},
+		ErrorKindNoKernelIOMMU: []Action{
+			ActionContactOSVendor, // suggest contacting the OS vendor to supply a kernel with this feature enabled.
+		},
+		ErrorKindTPMStartupLocalityNotProtected: []Action{
+			// TODO: Add an action to add PermitNoDiscreteTPMResetMitigation to CheckFlags
+		},
+		ErrorKindHostSecurity: []Action{
+			ActionContactOEM, // suggest contacting the OEM because of a firmware bug or misconfigured root-of-trust
+		},
+		ErrorKindPCRUnusable: []Action{
+			ActionContactOEM, // suggest contacting the OEM because of a firmware bug
+		},
+		ErrorKindVARSuppliedDriversPresent: []Action{
+			// TODO: Add action to add PermitVARSuppliedDrivers to CheckFlags if they're necessary - this gives the
+			// user the chance to be aware of their existence.
+			// TODO: If the drivers are being loaded from BDS using DriverOrder and DriverXXXX variables, add action to delete these
+		},
+		ErrorKindSysPrepApplicationsPresent: []Action{
+			// TODO: Add action to add PermitSysPrepApplications to CheckFlags if the user wants to keep them -
+			// this gives the user the chance to be aware of their existence.
+			// TODO: Add an action to just disable these by erasing the SysPrepOrder and SysPrepXXXX variables
+		},
+		ErrorKindAbsolutePresent: []Action{
+			ActionContactOEM,         // suggest contacting the OEM if there's no way to disable it.
+			ActionRebootToFWSettings, // suggest rebooting to the firmware settings UI to disable it.
+			// TODO: Add action to add PermitAbsoluteComputrace to CheckFlags if the user doesn't want to
+			// or can't disable it. This gives the user the chance to be aware of their existence.
+			// TODO: Add an action to just disable this automatically on supported platforms (eg, Dell via the WMI interface)
+		},
+		ErrorKindInvalidSecureBootMode: []Action{
+			ActionRebootToFWSettings, // suggest rebooting to the firmware settings UI to properly configure secure boot
+			// TODO: Add action to add PCRProfileOptionPermitNoSecureBootPolicyProfile to the PCRProfileOptionsFlags,
+			//  noting that this flag will need to be subsequently passed to WithAutoTCGPCRProfile.
+		},
+		ErrorKindWeakSecureBootAlgorithmsDetected: []Action{
+			// TODO: Add action to add PermitWeakSecureBootAlgorithms to CheckFlags.
+		},
+		ErrorKindPreOSDigestVerificationDetected: []Action{
+			// TODO: Add action to add PermitPreOSVerificationUsingDigests to CheckFlags.
+		},
+	}
+}
+
+// RunChecksContext maintains context for multiple invocations of [RunChecks] to permit the
+// install process to iterate and resolve detected issues where possible. It also reduces
+// the burden of selecting an initial set of [CheckFlags].
+type RunChecksContext struct {
+	env          internal_efi.HostEnvironment
+	flags        CheckFlags
+	loadedImages []secboot_efi.Image
+	profileOpts  PCRProfileOptionsFlags
+
+	errs    []error
+	lastErr error
+	result  *CheckResult
+
+	availableActions map[Action]bool
```

Comment:
this probably merits a comment as it seems it can store both true and false entries which is not always the case with such maps, sometimes they are really meant to be sets (so either no entry or true)

Reply:
This is intended to have 3 states - empty implies that we need to test that the action is available, false is "tested, but unavailable" and true is "available".

Reply:
I've added a doc comment for this now.